Mage/src/main/java/mage/abilities/costs/mana/ManaCostImpl.java

```diff
@@ -139,7 +139,7 @@ protected void assignColorless(Ability ability, Game game, ManaPool pool, int ma
         int conditionalCount = pool.getConditionalCount(ability, game, null, costToPay);
         while (mana > payment.count() && (pool.count() > 0 || conditionalCount > 0)) {
             if (pool.pay(ManaType.COLORLESS, ability, sourceFilter, game, costToPay, usedManaToPay)) {
-                this.payment.increaseColorless(1, pool.getLastPaymentWasSnow());
```

Comment:
Why you removed snow tracking from the `payment`? It must track it too (cause it mana object and must be synced).